src/price_data.py

```diff
@@ -391,8 +405,12 @@ def _get_price_kraken(
             if closest_match_index == -1:
                 continue
 
-            # The desired timestamp is in the future
-            if closest_match_index == len(data_timestamps_ms) - 1:
+            now_timestamp = misc.to_ms_timestamp(datetime.datetime.now().astimezone())
+            # The desired timestamp is in the future. Ignore if the target timestamp
+            # is very close to the current timestamp (for virtual sells)
+            if closest_match_index == len(data_timestamps_ms) - 1 and \
+                (now_timestamp > target_timestamp + 3600 * 1000
+                    or not config.CALCULATE_VIRTUAL_SELL):
```

Comment:
I find this really hard to read.

What do you think of something like this?
```
# Check if he have to adjust the fetched timestamp interval.
# We have to adjust, if the fetched timestamp is in the future.
timestamp_in_future = closest_match_index == len(data_timestamps_ms) - 1
# We do not have to adjust, for future prices, when we just want to calculate virtual sells.
accept_timestamp = config.CALCULATE_VIRTUAL_SELL and abs(now_timestamp - target_timestamp) < 60 * 60 * 1000
if timestamp_in_future and not accept_timestamp:
    ....
```

Why can we accept the fetched price for virtual sells? This price is written to the database and used next year for tax calculation. I don't understand why prices for virtual calculations can be fetched differently. Please explain your thoughts.

Reply:
My thought was that for virtual sells you just want to get the current price, irrelevant if it's before or after the timestamp. But you're right, I didn't consider that this price is then stored and could mix up future tax calculations (if a trade is done at the same time as the virtual sell timestamp).

Reply:
So the problem is that the received price data is only accepted if the target timestamp is between the timestamps of the received data points. For virtual sells, the target timestamp (basically the current time) is usually newer than the latest available trade, which leads to rejection of the whole dataset.
I've implemented a different approach now: The Kraken API returns the latest 1000 trades. If less than 100 trades are received, it is assumed that the latest trade price has been reached, which is then returned. There's also a check that issues a warning if the target timestamp is older than one hour in such a case.
What do you think of this approach?

Reply:
not perfect, but should be good enough for our use case